pkg/messaging/direct_messaging.go

```diff
@@ -262,13 +262,20 @@ func (d *directMessaging) invokeRemote(ctx context.Context, appID, namespace, ap
 }
 
 func (d *directMessaging) addDestinationAppIDHeaderToMetadata(appID string, req *invokev1.InvokeMethodRequest) {
-	req.Metadata()[invokev1.DestinationIDHeader] = &internalv1pb.ListStringValue{
+	metadata := req.Metadata()
```

Comment:
It is another issue, isn't it? I would create another PR to fix it. Also, do not forget add tests.

Reply:
ok, I check and repair it.